### PR TITLE
predefined Validation Functions

### PR DESCRIPTION
**Pull request by anirudhaCodes**
This PR refers to issue #134 .

### CODE REVIEW

File: validate.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"fmt"
+	"unicode/utf8"
+)
+
+// ValidateNotEmpty checks if the input is not empty.
+func ValidateNotEmpty() func(s string) error {
+	return func(s string) error {
+		if s == "" {
+			return fmt.Errorf("input cannot be empty")
```
Comment (caarlos0) [inline validate.go:12]: I think it would be nice to export these errors, so users can use `errors.Is` with them 🤔

File: validate.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// ValidateOneOf checks if a string is one of the specified options.
+func ValidateOneOf(options ...string) func(string) error {
+	validOptions := make(map[string]struct{})
+	for _, option := range options {
+		validOptions[option] = struct{}{}
+	}
+
+	return func(value string) error {
+		if _, ok := validOptions[value]; !ok {
+			return fmt.Errorf("invalid option")
```
Comment (caarlos0) [inline validate.go:38]:
```suggestion
			return fmt.Errorf("invalid option: %s", value)
```

File: validate.go (hunk 1 above)
Comment (anirudhaCodes) [reply validate.go:12]: Yes make sense.

I was also thinking that error message can be a input. Because the error message might depend on final use case.

**Comment by maaslalani** [comment]
Hey @anirudhaCodes thanks for this PR.

I haven't fully decided on the naming of these functions so they might change.

One thing so would suggest is to defined a `ValidateMinLength` function and a `ValidateMaxLength` function and then use these two functions to create a `ValidateLength` function and also Alias `ValidateNotEmpty` to `MinLength(1)`, etc...

That way we can make all the errors consistent and nicer and the functions easier to test.

File: validate.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+)
+
+// ErrInputEmpty is an error indicating that the input cannot be empty.
+var ErrInputEmpty = fmt.Errorf("input cannot be empty")
+
+// ValidateNotEmpty checks if the input is not empty.
+func ValidateNotEmpty() func(s string) error {
+	return func(s string) error {
+		if err := ValidateMinLength(1)(s); err != nil {
+			return ErrInputEmpty
```
Comment (maaslalani) [inline validate.go:15]:
```suggestion
			return fmt.Errorf("input cannot be empty")
```

File: validate.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package huh
+
+import (
+	"fmt"
+	"unicode/utf8"
+)
+
+// ErrInputEmpty is an error indicating that the input cannot be empty.
+var ErrInputEmpty = fmt.Errorf("input cannot be empty")
+
```
Comment (maaslalani) [inline validate.go:10]:
```suggestion
```

**Pull request merged**